diff --git a/Makefile.am b/Makefile.am
index ed319cd..2e620a4 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,3 +1,4 @@
+ACLOCAL_AMFLAGS = -I m4
 glib_enum_prefix = gst_dvbsink
 
 include glib-gen.mak
diff --git a/configure.ac b/configure.ac
index 8bf8c49..11be6c7 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,4 +1,6 @@
 AC_INIT
+AC_CONFIG_MACRO_DIR([m4])
+m4_include([m4/as-version.m4])
 
 dnl versions of gstreamer and plugins-base
 AC_ARG_WITH(gstversion,
