diff --git a/cmdutils.c b/cmdutils.c
index 386db3d..0e6ab81 100644
--- a/cmdutils.c
+++ b/cmdutils.c
@@ -623,8 +623,8 @@ static void print_program_info(int flags, int level)
         av_log(NULL, level, " Copyright (c) %d-%d the FFmpeg developers",
                program_birth_year, this_year);
     av_log(NULL, level, "\n");
-    av_log(NULL, level, "%sbuilt on %s %s with %s %s\n",
-           indent, __DATE__, __TIME__, CC_TYPE, CC_VERSION);
+    av_log(NULL, level, "%sbuilt with %s %s\n",
+           indent, CC_TYPE, CC_VERSION);
     av_log(NULL, level, "%sconfiguration: " FFMPEG_CONFIGURATION "\n", indent);
 }
 
diff --git a/ffprobe.c b/ffprobe.c
index ca6133e..f58e6a0 100644
--- a/ffprobe.c
+++ b/ffprobe.c
@@ -1648,8 +1648,10 @@ static void ffprobe_show_program_version(WriterContext *w)
     print_str("version", FFMPEG_VERSION);
     print_fmt("copyright", "Copyright (c) %d-%d the FFmpeg developers",
               program_birth_year, this_year);
+#if 0
     print_str("build_date", __DATE__);
     print_str("build_time", __TIME__);
+#endif
     print_str("compiler_type", CC_TYPE);
     print_str("compiler_version", CC_VERSION);
     print_str("configuration", FFMPEG_CONFIGURATION);
