--- ffmpeg-2.2/libavformat/mpegts.c	2014-03-23 23:08:02.000000000 +0100
+++ ffmpeg-2.2/libavformat/mpegts.c	2014-04-11 12:25:47.284959259 +0200
@@ -1500,7 +1500,15 @@
                     memcpy(extradata, *pp, 2);
                     extradata += 2;
 
-                    *pp += 2;
+
+		    char tmp_key[20];
+		    char tmp_val[40];
+		    AVDictionaryEntry *t = NULL;
+		    unsigned char b3 = get8(pp, desc_end);
+		    unsigned char b4 = get8(pp, desc_end);
+		    snprintf(tmp_key, sizeof(tmp_key), "teletext_%d", i);
+		    snprintf(tmp_val, sizeof(tmp_val), "%d %c%c%c %u %u %u", pid, language[i * 4], language[i * 4 + 1], language[i * 4 + 2], b3 >> 3, b3 & 0x7, b4); // pid lang type mag page
+		    av_dict_set(&st->metadata, tmp_key, tmp_val, 0);
                 }
 
                 language[i * 4 - 1] = 0;
