diff --git a/libavformat/mpegts.c b/libavformat/mpegts.c
index 8b92bc4..eb6ec23 100644
--- a/libavformat/mpegts.c
+++ b/libavformat/mpegts.c
@@ -1365,11 +1365,24 @@ int ff_parse_mpeg2_descriptor(AVFormatContext *fc, AVStream *st, int stream_type
         }
         break;
     case 0x56: /* DVB teletext descriptor */
-        language[0] = get8(pp, desc_end);
-        language[1] = get8(pp, desc_end);
-        language[2] = get8(pp, desc_end);
-        language[3] = 0;
-        av_dict_set(&st->metadata, "language", language, 0);
+	{
+		int ttx_count = desc_len / 5;
+		for (i = 0; i < ttx_count; i++) {
+			char tmp_key[20];
+			char tmp_val[40];
+			language[0] = get8(pp, desc_end);
+			language[1] = get8(pp, desc_end);
+			language[2] = get8(pp, desc_end);
+			language[3] = 0;
+			av_dict_set(&st->metadata, "language", language, 0);
+			AVDictionaryEntry *t = NULL;
+			unsigned char b3 = get8(pp, desc_end);
+			unsigned char b4 = get8(pp, desc_end);
+			snprintf(tmp_key, sizeof(tmp_key), "teletext_%d", i);
+			snprintf(tmp_val, sizeof(tmp_val), "%d %s %u %u %u", pid, language, b3 >> 3, b3 & 0x7, b4); // pid lang type mag page
+			av_dict_set(&st->metadata, tmp_key, tmp_val, 0);
+		}
+	}
         break;
     case 0x59: /* subtitling descriptor */
         language[0] = get8(pp, desc_end);
