diff -ru libcap-2.20/libcap/Makefile libcap-2.20-build//libcap/Makefile
--- libcap-2.20/libcap/Makefile	2010-08-09 00:26:04.000000000 +0200
+++ libcap-2.20-build//libcap/Makefile	2011-03-08 20:59:23.000000000 +0100
@@ -16,7 +16,7 @@
 # no support).
 ifeq ($(LIBATTR),yes)
 FILES += cap_file
-LDFLAGS += -lattr
+LDFLAGS += -lattr -L$(TARGETPREFIX)/lib
 endif
 
 INCLS=libcap.h cap_names.h $(INCS)
@@ -33,7 +33,7 @@
 endif
 
 _makenames: _makenames.c cap_names.list.h
-	$(BUILD_CC) $(BUILD_CFLAGS) $< -o $@
+	$(HOST_CC) $(BUILD_CFLAGS) $< -o $@
 
 cap_names.h: _makenames
 	./_makenames > cap_names.h
@@ -68,9 +68,9 @@
 	install -m 0644 $(MINLIBNAME) $(LIBDIR)/$(MINLIBNAME)
 	ln -sf $(MINLIBNAME) $(LIBDIR)/$(MAJLIBNAME)
 	ln -sf $(MAJLIBNAME) $(LIBDIR)/$(LIBNAME)
-ifeq ($(FAKEROOT),)
-	-/sbin/ldconfig
-endif
+#ifeq ($(FAKEROOT),)
+#	-/sbin/ldconfig
+#endif
 
 clean:
 	$(LOCALCLEAN)
diff -ru libcap-2.20/Make.Rules libcap-2.20-build//Make.Rules
--- libcap-2.20/Make.Rules	2011-01-19 06:06:44.000000000 +0100
+++ libcap-2.20-build//Make.Rules	2011-03-08 20:58:14.430329802 +0100
@@ -47,12 +47,19 @@
 KERNEL_HEADERS := $(topdir)/libcap/include
 IPATH += -fPIC -I$(topdir)/libcap/include -I$(KERNEL_HEADERS)
 
-CC := gcc
+#CC := gcc
+HOST_CC := gcc
+CC := $(CROSS_BASE)/bin/$(TARGET)-gcc
+
 CFLAGS := -O2
 BUILD_CC := $(CC)
 BUILD_CFLAGS := $(CFLAGS) $(IPATH)
-AR := ar
-RANLIB := ranlib
+#AR := ar
+AR := $(CROSS_BASE)/bin/$(TARGET)-ar
+
+#RANLIB := ranlib
+RANLIB := $(CROSS_BASE)/bin/$(TARGET)-ranlib
+
 DEBUG = -g #-DDEBUG
 WARNINGS=-Wall -Wwrite-strings \
         -Wpointer-arith -Wcast-qual -Wcast-align \
@@ -61,7 +68,9 @@
 LD=$(CC) -Wl,-x -shared
 LDFLAGS := #-g
 
-SYSTEM_HEADERS = /usr/include
+#SYSTEM_HEADERS = /usr/include
+SYSTEM_HEADERS = $(TARGETPREFIX)/include
+
 INCS=$(topdir)/libcap/include/sys/capability.h
 LDFLAGS += -L$(topdir)/libcap
 CFLAGS += -Dlinux $(WARNINGS) $(DEBUG)
diff -ru libcap-2.20/progs/Makefile libcap-2.20-build//progs/Makefile
--- libcap-2.20/progs/Makefile	2010-08-09 01:56:40.000000000 +0200
+++ libcap-2.20-build//progs/Makefile	2011-03-08 20:58:14.000000000 +0100
@@ -14,7 +14,7 @@
 ifneq ($(DYNAMIC),yes)
 LDFLAGS += --static
 endif
-LDLIBS += -L../libcap -lcap
+LDLIBS += -L../libcap -lcap -Wl,--rpath -Wl,$(TARGETPREFIX)/lib
 
 all: $(BUILD)
 
