--- a/vsf_findlibs.sh
+++ b/vsf_findlibs.sh
@@ -1,8 +1,8 @@
 #!/bin/sh
 # Cheesy hacky location of additional link libraries.
 
-locate_library() { [ ! "$1*" = "`echo $1*`" ]; }
-find_func() { egrep $1 $2 >/dev/null; }
+locate_library() { [ ! "${TARGETPREFIX}$1*" = "`echo ${TARGETPREFIX}$1*`" ]; }
+find_func() { egrep ${TARGETPREFIX}$1 $2 >/dev/null; }
 
 if find_func hosts_access tcpwrap.o; then
   echo "-lwrap";
@@ -34,7 +34,7 @@ locate_library /lib/libdl.so && echo "-ldl";
 locate_library /lib/libsocket.so && echo "-lsocket";
 
 # Look for libnsl. Solaris needs this.
-locate_library /lib/libnsl.so && echo "-lnsl";
+# locate_library /lib/libnsl.so && echo "-lnsl";
 
 # Look for libresolv. Solaris needs this.
 locate_library /lib/libresolv.so && echo "-lresolv";
@@ -47,9 +47,9 @@ locate_library /usr/lib/libsec.sl && echo "-lsec";
 
 # Look for libcap (capabilities)
 if locate_library /lib/libcap.so.1; then
-  echo "/lib/libcap.so.1";
+  echo "${TARGETPREFIX}/lib/libcap.so.1";
 elif locate_library /lib/libcap.so.2; then
-  echo "/lib/libcap.so.2";
+  echo "${TARGETPREFIX}/lib/libcap.so.2";
 else
   locate_library /usr/lib/libcap.so && echo "-lcap";
   locate_library /lib/libcap.so && echo "-lcap";
