diff --git a/vtuner-dvb-3.c b/vtuner-dvb-3.c
index ec92e6d..78b2267 100644
--- a/vtuner-dvb-3.c
+++ b/vtuner-dvb-3.c
@@ -41,11 +41,15 @@ int hw_init(vtuner_hw_t* hw, int adapter, int frontend, int demux, int dvr) {
       #if DVB_API_VERSION < 5
         hw->type = VT_S;
       #else
+#ifdef __sh__
+        hw->type = VT_S2;
+#else
         if( hw->fe_info.caps & ( FE_HAS_EXTENDED_CAPS | FE_CAN_2G_MODULATION ) ) {
           hw->type = VT_S2; 
         } else {
           hw->type = VT_S;
         } 
+#endif
       #endif
       break;
     case FE_QAM:  hw->type = VT_C; break;
@@ -136,7 +140,7 @@ int hw_get_frontend(vtuner_hw_t* hw, struct dvb_frontend_parameters* fe_params)
   int ret;
   ret = ioctl(hw->frontend_fd, FE_GET_FRONTEND, fe_params);
   if( ret != 0 ) {
-    WARN(MSG_NET, "FE_GET_FRONTEND failed. It seems your DVB driver has incomplete implementation.\n");
+    DEBUG(MSG_NET, "FE_GET_FRONTEND failed (%m).\n");
     // Follows workaround for crappy drivers which have not implemented .get_frontend() callback
     memset(fe_params, 0, sizeof(struct dvb_frontend_parameters));
     ret = 0;
diff --git a/vtuner-utils.h b/vtuner-utils.h
index 73e1e65..e53c3d8 100644
--- a/vtuner-utils.h
+++ b/vtuner-utils.h
@@ -26,10 +26,10 @@ extern int use_syslog;
 #define MSG_INFO	3
 #define MSG_DEBUG	4
 
-#define ERROR(mtype, msg, ...) write_message(mtype, MSG_ERROR, "[%d %s:%u] error: " msg, getpid(), __FILE__, __LINE__, ## __VA_ARGS__)
-#define  WARN(mtype, msg, ...) write_message(mtype, MSG_WARN,  "[%d %s:%u]  warn: " msg, getpid(), __FILE__, __LINE__, ## __VA_ARGS__)
-#define  INFO(mtype, msg, ...) write_message(mtype, MSG_INFO,  "[%d %s:%u]  info: " msg, getpid(), __FILE__, __LINE__, ## __VA_ARGS__)
-#define DEBUG(mtype, msg, ...) write_message(mtype, MSG_DEBUG, "[%d %s:%u] debug: " msg, getpid(), __FILE__, __LINE__, ## __VA_ARGS__)
+#define ERROR(mtype, msg, ...) write_message(mtype, MSG_ERROR, "[%d %s:%u] error: " msg, getpid(), __func__, __LINE__, ## __VA_ARGS__)
+#define  WARN(mtype, msg, ...) write_message(mtype, MSG_WARN,  "[%d %s:%u]  warn: " msg, getpid(), __func__, __LINE__, ## __VA_ARGS__)
+#define  INFO(mtype, msg, ...) write_message(mtype, MSG_INFO,  "[%d %s:%u]  info: " msg, getpid(), __func__, __LINE__, ## __VA_ARGS__)
+#define DEBUG(mtype, msg, ...) write_message(mtype, MSG_DEBUG, "[%d %s:%u] debug: " msg, getpid(), __func__, __LINE__, ## __VA_ARGS__)
 
 void write_message(const unsigned int, const int, const char*, ...);
 int open_udplog(char *, int );
